src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -158,6 +183,822 @@ MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree)
 
 
 /*
+ * MultiSubqueryPlanTree gets the query objects and returns logical plan
+ * for subqueries.
+ *
+ * We currently have two different code paths for creating logic plan for subqueries:
+ *   (i) subquery pushdown
+ *   (ii) single relation repartition subquery
+ *
+ * In order to create the logical plan, we follow the algorithm below:
+ *    -  If subquery pushdown planner can plan the query
+ *        -  We're done, we create the multi plan tree and return
+ *    -  Else
+ *       - If the query is not eligible for single table repartition subquery planning
+ *            - Throw the error that the subquery pushdown planner generated
+ *       - If it is eligible for single table repartition subquery planning
+ *            - Check for the errors for single table repartition subquery planning
+ *                - If no errors found, we're done. Create the multi plan and return
+ *                - If found errors, throw it
+ */
+static MultiNode *
+MultiSubqueryPlanTree(Query *originalQuery, Query *queryTree,
+					  PlannerRestrictionContext *plannerRestrictionContext)
+{
+	MultiNode *multiQueryNode = NULL;
+	DeferredErrorMessage *subqueryPushdownError = NULL;
+
+	/*
+	 * This is a generic error check that applies to both subquery pushdown
+	 * and single table repartition subquery.
+	 */
+	ErrorIfQueryNotSupported(originalQuery);
+
+	/*
+	 * In principle, we're first trying subquery pushdown planner. If it fails
+	 * to create a logical plan, continue with trying the single table
+	 * repartition subquery planning.
+	 */
+	subqueryPushdownError = DeferErrorIfUnsupportedSubqueryPushdown(originalQuery,
+																	plannerRestrictionContext);
+	if (!subqueryPushdownError)
+	{
+		multiQueryNode = SubqueryPushdownMultiPlanTree(originalQuery);
+	}
+	else if (subqueryPushdownError)
+	{
+		bool singleRelationRepartitionSubquery =
```

Comment:
I would move this to before if block where we use the `singleRelationRepartitionSubquery`.